diff --git a/tensorflow_examples/models/pix2pix/pix2pix.py b/tensorflow_examples/models/pix2pix/pix2pix.py
index 1cfc1c8..163ed3f 100644
--- a/tensorflow_examples/models/pix2pix/pix2pix.py
+++ b/tensorflow_examples/models/pix2pix/pix2pix.py
@@ -25,6 +25,7 @@
 from absl import flags
 
 import tensorflow as tf
+import wandb
 
 FLAGS = flags.FLAGS
 
@@ -493,6 +494,7 @@
 def main(epochs, enable_function, path, buffer_size, batch_size):
   path_to_folder = path
 
+  wandb.init()
   pix2pix_object = Pix2pix(epochs, enable_function)
 
   train_dataset, _ = create_dataset(
