diff --git a/tensorflow_examples/models/densenet/train.py b/tensorflow_examples/models/densenet/train.py
index 5696331..37a15a3 100644
--- a/tensorflow_examples/models/densenet/train.py
+++ b/tensorflow_examples/models/densenet/train.py
@@ -22,6 +22,7 @@
 import tensorflow as tf
 from tensorflow_examples.models.densenet import densenet
 from tensorflow_examples.models.densenet import utils
+import wandb
 
 
 class Train(object):
@@ -60,10 +61,12 @@
   def keras_fit(self, train_dataset, test_dataset):
     self.model.compile(
         optimizer=self.optimizer, loss=self.loss_object, metrics=['accuracy'])
+    callbacks = []
+    callbacks.append(tf.keras.callbacks.LearningRateScheduler(self.decay))
+    callbacks.append(wandb.keras.WandbCallback())
     history = self.model.fit(
         train_dataset, epochs=self.epochs, validation_data=test_dataset,
-        verbose=2, callbacks=[tf.keras.callbacks.LearningRateScheduler(
-            self.decay)])
+        verbose=2, callbacks=callbacks)
     return (history.history['loss'][-1],
             history.history['accuracy'][-1],
             history.history['val_loss'][-1],
@@ -177,6 +180,7 @@
          train_mode='custom_loop',
          data_dir=None):
 
+  wandb.init()
   model = densenet.DenseNet(mode, growth_rate, output_classes, depth_of_model,
                             num_of_blocks, num_layers_in_each_block,
                             data_format, bottleneck, compression, weight_decay,
